### PR TITLE
EC Analytics

### PR DESCRIPTION
**Pull request by ericthewry**
Hey @sf-interns/owners, here is some of the basic Google Analytics setup.

If you want to test to see how this works and what information it is giving us, you can use the [Google Analytics Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) to see the information.  I created a google account for our the analytics data. You can log into the [Analytics page](https://www.google.com/analytics/) to see the data.

Here's the information we'll get:
- Every time someone accesses the page, we will get a `pageview` event with the parameter `Mobile` or `Desktop` based on the access method.
- We'll get `event`s tied to the `board` on `reset`, `new game`, `initialize` and `solved`.  Using some GA metrics. We'll be able to assess which patterns of use are most common, and which become outmoded as we progress with future features.
- Are there any other events we want?

I'm going to do a refactor so that we have a `TrackingService` through which all of the GA events go.  It will prevent us from having to chase down silly strings in the code base.

Let me know if you have questions, comments, etc.

EDIT: `TrackingService` completed.
BW EDIT: Remove email so no one public can see it.

### CODE REVIEW

File: app/services/TrackingService.coffee [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @pageview 'Desktop'
+
+  @pageview: (label) ->
+    if label?
+      ga 'send', 'pageview', label
+    else
+      ga 'send', 'pageview'
+
+
+module.exports = TrackingService
```
Comment (bmwwoo) [inline app/services/TrackingService.coffee]: newline and extra whitespace

Comment (ericthewry) [inline app/services/TrackingService.coffee]: :+1:

**Comment by bmwwoo** [comment]
@sf-interns/owners Do you think we should start squash merging commits? Otherwise LGTM

**Comment by ericthewry** [comment]
I have no preference about squashing for this project.  

Typically I only squash when I merge _into_ a feature branch.

File: app/services/TrackingService.coffee [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  @desktopView: ->
+    @pageview 'Desktop'
+
+  @pageview: (label) ->
+    if label?
+      ga 'send', 'pageview', label
+    else
+      ga 'send', 'pageview'
+
+module.exports = TrackingService
```
Comment (chuj) [inline app/services/TrackingService.coffee]: new line

**Comment by chuj** [comment]
awesome! resolve merge conflicts and then LGTM :+1:

**Comment by bmwwoo** [comment]
Fix adblock, ghostery, and ublock preferably before you merge in

**Comment by ericthewry** [comment]
@chuj Merge Conflict?

**Pull request merged**